SolidCP/Sources/SolidCP.Providers.TerminalServices.Windows2012/Windows2012.cs

```diff
@@ -1341,11 +1341,18 @@ private void SetRdsSessionHostPermissions(Runspace runspace, RdsServerSettings s
    
         private void RemoveRegistryValue(Runspace runspace, string key, string gpoName)
         {
-            Command cmd = new Command("Remove-GPRegistryValue");
-            cmd.Parameters.Add("Name", gpoName);
-            cmd.Parameters.Add("Key", string.Format("\"{0}\"", key));
+            try
+            {
+                Command cmd = new Command("Remove-GPRegistryValue");
+                cmd.Parameters.Add("Name", gpoName);
+                cmd.Parameters.Add("Key", string.Format("\"{0}\"", key));
 
-            Collection<PSObject> result = runspace.ExecuteRemoteShellCommand(PrimaryDomainController, cmd, PrimaryDomainController);
+                Collection<PSObject> result = runspace.ExecuteRemoteShellCommand(PrimaryDomainController, cmd, PrimaryDomainController);
+            }
+            catch (Exception e)
```

Comment:
Hi, Are you sure the ExecuteRemoteShellCommand throw an Exception?
Cause this https://github.com/FuseCP/SolidCP/blob/028c473f39ae19af6b75edc315d84250a88343e3/SolidCP/Sources/SolidCP.Providers.TerminalServices.Windows2012/RdsRunspaceExtensions.cs#L262 online write errors, but not throws it like exceptions.
You probably need to modify the ExecuteRemoteShellCommand like it is there: https://github.com/FuseCP/SolidCP/blob/028c473f39ae19af6b75edc315d84250a88343e3/SolidCP/Sources/SolidCP.Providers.Virtualization.HyperV-2012R2/PowerShellManager.cs#L184

Reply:
for me "being sure" is a crucial aspect in human communication. I wrote this code months ago so memory-wise it is all a bit vague and I don't have the time to try and reproduce it. So in that sense I'm not sure. 
But I don't tend to write code without purpose ;-)

Yes, ExecuteShellCommand does not throw errors, and only writes errors, but that does not mean that nothing can go wrong there. In my experience it does go wrong if the GPO key does not exist. 

Yes, a global Try catch in ExecuteRemoteShellCommand is also an option. 
Is that a better option?

Reply:
From my experience with PowerShell in the HyperV2012 part of code. This catch probably will never be executed, but it is HyperV, not TermianlServrices, so it was just my five cents :)